### PR TITLE
Super scope patch - fix issue #45

### PR DESCRIPTION
**Pull request by SSoulaimane**

### CODE REVIEW

**Comment by SSoulaimane** [comment]
test utils pr #42 should be mergred before this.

**Comment by SSoulaimane** [comment]
ref issue #45

File: src/dsymbol/tests.d [hunk 1]
```diff
@@ -22,18 +25,12 @@ void expectSymbolsAndTypes(const string source, const string[][] results,
...
-
     ModuleCache mcache = ModuleCache(theAllocator);
-    mcache.cacheModule(fName);
+    auto pair = generateAutocompleteTrees(source, mcache);
+    scope(exit) pair.destroy();
 
     size_t i;
-    foreach (const(CacheEntry)* s; mcache.getAllSymbols)
-        foreach(ss; (*s.symbol)[])
+    foreach(ss; (*pair.symbol)[])
```
Comment (BBasile) [inline src/dsymbol/tests.d]: `pair` should be checked before dereference

Comment (BBasile) [reply src/dsymbol/tests.d]: well no actually...

**Pull request merged**